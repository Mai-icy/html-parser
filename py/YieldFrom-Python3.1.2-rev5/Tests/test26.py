#
#   Test throwing GeneratorExit into a subgenerator that
#   catches it and returns normally.
#

def f():
	try:
		print("Enter f")
		yield
		print("Exit f")
	except GeneratorExit:
		return

def g():
	print("Enter g")
	yield from f()
	print("Exit g")

gi = g()
next(gi)
gi.throw(GeneratorExit)
