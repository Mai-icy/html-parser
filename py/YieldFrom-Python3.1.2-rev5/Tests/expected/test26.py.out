Enter g
Enter f
Traceback (most recent call last):
  File "test26.py", line 21, in <module>
    gi.throw(GeneratorExit)
  File "test26.py", line 16, in g
    yield from f()
GeneratorExit
